lib/services/publish-service.ts

```diff
@@ -0,0 +1,208 @@
+            } catch (err) {
+                this.$logger.printInfoMessageOnSameLine(os.EOL);
+                if (err.message === PublishService.ERROR_MESSAGE_WHEN_PROJECT_EXISTS_ON_REMOTE) {
+                    this.$errors.failWithoutHelp("This project already exists on the remote server. " + os.EOL + 
+                                    "To remove all files and upload the new ones on the remote run $ appbuilder publish remove --force");
+                }
+
+                throw err;
+            }
+
+            this.$logger.printInfoMessageOnSameLine(os.EOL);
+            this.$logger.info("Project '%s' published successfully.", projectName);
+        }).future<void>()();
+    }
+
+    public listAllConnections(): IFuture<void> {
+        return (() => {
+            this.listConnections(this.ftpPublishConnections);
+        }).future<void>()();
+    }
+
+    public addConnection(name: string, publishUrl: string): IFuture<void> {
+        return (() => {
+            var ftpPublishConnection: IPublishConnection = {
+                type: PublishService.ALLOWED_CONNECTION_TYPE,
+                publicUrl: PublishService.PUBLIC_URL_DEFAULT_VALUE,
+                publishUrl: publishUrl,
+                name: name
+            };
+
+            if (!ftpPublishConnection.name) {
+                ftpPublishConnection.name = this.$prompter.getString("Name:").wait();
+            }
+
+            if (!ftpPublishConnection.publishUrl) {
+                ftpPublishConnection.publishUrl = this.$prompter.getString("Publish URL:").wait();
+            }
+
+            this.allPublishConnections.push(ftpPublishConnection);
+            this.savePublishJsonFile().wait();
+            this.$logger.info("Connection '%s' added successfully.", ftpPublishConnection.name);
+        }).future<void>()();
+    }
+
+    public removeConnection(idOrName: string): IFuture<void> {
+        return (() => {
+            var id = parseInt(idOrName) - 1;
+            var message : string;
+            if (!isNaN(id)) {
+                this.validateId(id);
+
+                var connectionIndex = _.findIndex(this.allPublishConnections, this.ftpPublishConnections[id]);
+                this.allPublishConnections.splice(connectionIndex, 1);
+                message = util.format("Connection '%s' successfully removed.", this.ftpPublishConnections[id].name);
+            } else {
+                var name = idOrName;
+                var connectionsToBeRemoved = _.filter(this.ftpPublishConnections, c => c.name === name);
+
+                if (connectionsToBeRemoved.length > 1) {
+                    var connectionNameUrls = _.map(connectionsToBeRemoved, c => c.name + '(' + c.publishUrl + ')');
+                    connectionNameUrls.splice(connectionNameUrls.length, 0, PublishService.REMOVE_ALL_CONNECTIONS_MESSAGE);
```

Comment:
maybe `push` or `unshift` (based on which position you think it is better to add the option) as it is more readable for me (this is just a suggestion)